library/std/src/process.rs

```diff
@@ -2030,6 +2030,11 @@ pub fn id() -> u32 {
 ///
 /// The default implementations are returning `libc::EXIT_SUCCESS` to indicate
 /// a successful execution. In case of a failure, `libc::EXIT_FAILURE` is returned.
+///
+/// Note that this trait is supposed to be used in standard library runtime,
+/// where returned value from main function is considered meaningful.
+/// On the other hand, implementing Termination in no-std environment, where
+/// there is no notion of `exit status`, has no effect or may possibly cause bugs.
```

Comment:
The issue isn't that it could lead to bugs, `std::process::Termination` and `std::process::exit` wouldn't be available in a `no_std` binary so there would be no way to misuse it. I believe the issue that @Lokathor wanted to clarify was that `Termination` is intended to be a piece of the standard library's runtime that is provided for convenience and should not be seen as a general requirement on what other runtime implementations must do.

Reply:
Yes. My concern is not that implementing `Termination` is ever an error/bug. Instead, I would just like the docs to be clear that *how the value is interpreted* when it's returned from `main` is not something to rely on in alternative runtimes.